contracts/ERC20.sol

```diff
@@ -2,33 +2,208 @@
 pragma solidity ^0.8.0;
 
 import "./IERC20.sol";
+import "./IERC20Metadata.sol";
 
 contract ERC20 is IERC20 {
-  function totalSupply() external view returns (uint256) {
-    return 0;
-  }
 
-  function balanceOf(address account) external view returns (uint256) {
-    return 0;
-  }
 
-  function transfer(address to, uint256 amount) external returns (bool) {
-    return true;
-  }
+   /*///////////////////////////////////////////////////////////////
+                        State Variables:
+    //////////////////////////////////////////////////////////////*/
+    uint256 private totalSupply_;
+    uint8 private decimals_ = 18;
+    uint256 public maxSupplyMintPerWallet = 10**decimals_;
+    uint256 public maxSupply = 100*(10**decimals_); 
+    
+    string private name_;
+    string private symbol_;
 
-  function allowance(address owner, address spender) external view returns (uint256) {
-    return 0;
-  }
+    address private admin_;
+
+     /*///////////////////////////////////////////////////////////////
+                        Mappings:
+    //////////////////////////////////////////////////////////////*/
+    mapping(address => uint256) private balances;
+
+    mapping(address => mapping(address => uint)) private allowances;
+
+    mapping(address => uint256) private userMintTracker;
+
+
+         /*///////////////////////////////////////////////////////////////
+                        Events:
+    //////////////////////////////////////////////////////////////*/
+    event AdminTransfer(address oldAdmin, address newAdmin);
+
+     /*///////////////////////////////////////////////////////////////
+                          Modifiers
+    //////////////////////////////////////////////////////////////*/
+
+    modifier onlyAdmin() {
+      require(msg.sender == admin_, "VTZY: Unauthorised user.");
+      _;
+    }
+
+    modifier safeMintAmount(uint256 amount) {
+      require(userMintTracker[msg.sender] + amount <= maxSupplyMintPerWallet, "VTZY: Total minting amount exceeds limit.");
+      require(totalSupply_ + amount <= maxSupply, "VTZY: Total supply exceeds max supply.");
+      _;
+    }
+
+         /*///////////////////////////////////////////////////////////////
+                          Constructor
+    //////////////////////////////////////////////////////////////*/
+    constructor(string memory _name, string memory _symbol) {
+      name_ = _name;
+      symbol_ = _symbol;
+      admin_ = msg.sender;
+      _mint(msg.sender, 1 ether);
```

Comment:
1 ether is usually used for actual 1 ether. For other circumstances, we simply use 10**18